src/engine/Resources/Sound/AudioContext.ts

```diff
@@ -0,0 +1,100 @@
+export class AudioContextFactory {
+   public static create() {
+      if ((<any>window).AudioContext || (<any>window).webkitAudioContext) {
```

Comment:
I'm fairly certain we want only 1, we tried using multiple `AudioContext`s in the past. At the time browsers did not handle that well. 

A single `AudioContext` can play multiple concurrent audio streams, so there is no reason to support multiple.

Reply:
What would a `OfflineAudioContext` do?

Reply:
Typically  `OfflineAudioContext` is used to pre-process audio data and than pass it to `AudioBuffer`. One of use-cases is described here:
When you want to create real-time canvas visualization (https://stackoverflow.com/questions/8074152/is-there-a-way-to-use-the-web-audio-api-to-sample-audio-faster-than-real-time).

and about multiple `AudioContexts`, maybe, we could create a pending issue, waiting for better support in browsers? Because `AudioContext` has useful methods like `suspend`, `resume`, etc. it could be useful while creating sound collections. Particularly, we could use multiple `AudioContext`s for issue #590 . Dividing sound in groups like "music", "ambient sounds", "sfx" etc. Handling each group in bounds of separate `AudioContext` would be pretty neat, i think.